pkg/ccl/crosscluster/logical/dead_letter_queue.go

```diff
@@ -29,81 +57,93 @@ const (
 func (t WriteType) String() string {
 	switch t {
 	case Insert:
-		return "Insert"
+		return "insert"
 	case Delete:
-		return "Delete"
+		return "delete"
 	default:
 		return fmt.Sprintf("Unrecognized WriteType(%d)", int(t))
 	}
 }
 
-type ConflictResolutionType int
-
-const (
-	Unresolved ConflictResolutionType = iota
-	ManuallyResolved
-	Applied
-	Ignored
-)
-
 type DeadLetterQueueClient interface {
-	Create() error
+	Create(
+		ctx context.Context, jobExecCtx sql.JobExecContext, tableIDs []int32,
+	) error
 
 	Log(
 		ctx context.Context,
+		ie isql.Executor,
 		ingestionJobID int64,
 		kv streampb.StreamEvent_KV,
 		cdcEventRow cdcevent.Row,
-		reason error,
+		dlqReason retryEligibility,
 	) error
 }
 
-type loggingDeadLetterQueueClient struct {
+type deadLetterQueueClient struct {
 }
 
-// TODO(azhu): Implement after V0.
-// In future iterations Create() should create a DLQ table in the database.
-func (dlq *loggingDeadLetterQueueClient) Create() error {
+func (dlq *deadLetterQueueClient) Create(
+	ctx context.Context, jobExecCtx sql.JobExecContext, tableIDs []int32,
+) error {
+	db := jobExecCtx.ExecCfg().InternalDB.Executor()
+
+	if _, err := db.Exec(ctx, "create-write-type-enum", nil, writeEnumStmt); err != nil {
+		return errors.Wrap(err, "failed to create write enum")
+	}
+
+	// Create a dlq table for each given table.
+	for _, tableID := range tableIDs {
+		tableName := fmt.Sprintf(dlqBaseTableName, tableID)
+		if _, err := db.Exec(ctx, "create-dlq-table", nil, fmt.Sprintf(createTableStmt, tableName)); err != nil {
+			return errors.Wrapf(err, "failed to create dlq for table %s", tableName)
+		}
+	}
 	return nil
 }
 
-// TODO(azhu): Append complete log as rows to DLQ table.
-func (dlq *loggingDeadLetterQueueClient) Log(
+func (dlq *deadLetterQueueClient) Log(
 	ctx context.Context,
+	ie isql.Executor,
 	ingestionJobID int64,
 	kv streampb.StreamEvent_KV,
 	cdcEventRow cdcevent.Row,
-	reason error,
+	dlqReason retryEligibility,
 ) error {
-	if !dlq.exists() {
-		return errors.New("dead letter queue table needs to be created before logs can be appended")
-	}
-
 	if !cdcEventRow.IsInitialized() {
 		return errors.New("cdc event row not initialized")
 	}
 
 	tableID := cdcEventRow.TableID
-	var writeType WriteType
+	tableName := fmt.Sprintf(dlqBaseTableName, tableID)
```

Comment:
[minor] Not a big deal, but if we put something like a `insertStmts map[catpb.DescID]string` on the deadLetterQueueClient, we could create avoid the to Sprintf's in this method since they should be the same every time.  It'll be a drop in the bucket if you go with my suggestion of marshalling the whole stream event though.

Reply:
Will address this in a followup pr